sdk/javascript/processor/exceptions.js

```diff
@@ -17,16 +17,31 @@
 
 'use strict'
 
-class InvalidTransaction extends Error {
-  constructor (message = '') {
+class _TransactionProcessorError extends Error {
+  constructor (message = '', extendedData = null) {
```

Comment:
The data is optional, extended application data.  It should be marked as such here.  Added a doc string to reflect this fact.